builtin/logical/aws/client.go

```diff
@@ -36,22 +36,21 @@ func getRootConfig(ctx context.Context, s logical.Storage, clientType string, lo
 
 		credsConfig.AccessKey = config.AccessKey
 		credsConfig.SecretKey = config.SecretKey
```

Comment:
This seems like it would change behavior for configurations that explicitly set the [region](https://developer.hashicorp.com/vault/api-docs/secret/aws#region) and don't set [iam_endpoint](https://developer.hashicorp.com/vault/api-docs/secret/aws#iam_endpoint) or [sts_endpoint](https://developer.hashicorp.com/vault/api-docs/secret/aws#sts_endpoint)?

Reply:
Uploaded the following commit to address this: https://github.com/hashicorp/vault/pull/22726/commits/51aee8e51d6388da2c83d2e12e0a432bd1d0a6d8